src/services/formatting/smartIndenter.ts

```diff
@@ -400,5 +415,16 @@ module ts.formatting {
                     return false;
             }
         }
+
+        export function isPassableBlockForm(kind: SyntaxKind): boolean {
+            switch (kind) {
+                case SyntaxKind.ArrowFunction:
+                case SyntaxKind.FunctionExpression:
+                case SyntaxKind.ArrayLiteralExpression:
+                case SyntaxKind.ObjectLiteralExpression:

```

Comment:
I don't understand; `getContainingList` already covers all these cases. Perhaps the logic should be moved into there.